### PR TITLE
GH-956: Review recent PRs.

### PR DESCRIPTION
**Pull request by tbl3rd**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-956

### Changes

- No changes.

### Review Instructions

- No instructions.

### CODE REVIEW

File: src/ptc/util/jms.clj [hunk 1]
```diff
@@ -213,13 +206,16 @@
...
+    (letfn [(missing? [k] (when (nil? (k workflow)) k))
+            (none? [one-of]
+              (when (not-any? one-of (keys workflow))
+                one-of))]
+      (let [missing (concat (keep missing? required) (keep none? one-ofs))]
+        (when (seq missing)
+          (throw (IllegalArgumentException.
+                  (str/join \space [missing-keys-message (vec missing)])))))
+      (let [params (push-params prefix workflow)]
+        (push-append-to-aou-request prefix workflow params)))))
```
Comment (ehigham) [inline src/ptc/util/jms.clj:221]: I don't really get `push-append-to-aou-request` and I'm not sure changing its signature helps...
I'd feel a bit more comfortable if it did only send a request and we returned the tuple here:
```
(let [params (push-params prefix workflow)
      ptc    (str (cloud-prefix prefix workflow) "/ptc.json")]
  (push-append-to-aou-request prefix workflow params ptc)
  [params ptc])
```

File: src/ptc/util/misc.clj [hunk 2]
```diff
@@ -138,7 +138,8 @@
     [bucket (or object "")]))
 
 (defn gcs-object-exists?
-  "Return FILE stat if the gcs object exists."
-  [file]
-  (do-or-nil
-   (shell! "gsutil" "stat" file)))
+  "Return PATH when there is a GCS object at PATH.  Otherwise nil."
+  [path]
+  (when (string? path)
```
Comment (ehigham) [inline src/ptc/util/misc.clj:143]: is `nil` a valid argument?

File: src/ptc/util/jms.clj [hunk 3]
```diff
@@ -79,23 +77,15 @@
...
-  [keys push workflow]
-  (let [[key & rest] keys]
-    (if key
-      (do (let [updated-keys (update-cloud-path-keys key push workflow)]
-            (handle-existing-cloud-paths rest updated-keys workflow)))
-      push)))
+(defn wfl-keys->jms-keys-for
+  "Return wfl-keys->jms-keys modified for WORKFLOW."
+  [workflow]
+  (letfn [(frob [keymap [k v]] (assoc-in keymap [::push k] v))]
```
[the remark this answers is not in the document]
Comment (tbl3rd) [reply src/ptc/util/jms.clj:83]: Change `frob` to `adapt`?

File: src/ptc/util/jms.clj [hunk 4]
```diff
@@ -164,31 +151,37 @@
...
-  [workflow]
-  (let [aou-reference-bucket (or (System/getenv "AOU_REFERENCE_BUCKET") "broad-arrays-dev-storage")
-        cloud-chip-metadata-dir (get workflow :cloudChipMetaDataDirectory)
-        bucket (first (misc/parse-gs-url cloud-chip-metadata-dir))
-        aou-chip-metadata-dir (str/replace-first cloud-chip-metadata-dir bucket aou-reference-bucket)
-        extended-chip-manifest-file-name (get workflow :extendedIlluminaManifestFileName)]
-    (str aou-chip-metadata-dir extended-chip-manifest-file-name)))
+  "Get the extended_chip_manifest_file from WORKFLOW."
+  [{:keys [cloudChipMetaDataDirectory extendedIlluminaManifestFileName]
+    :as workflow}]
```
[the remark this answers is not in the document]
Comment (tbl3rd) [reply src/ptc/util/jms.clj:162]: Which?  `workflow`?

File: src/ptc/util/misc.clj (hunk 2 above)
Comment (tbl3rd) [reply src/ptc/util/misc.clj:143]: Yes, or at least it is not an invalid one.

File: src/ptc/util/jms.clj (hunk 1 above)
Comment (tbl3rd) [reply src/ptc/util/jms.clj:221]: I'll call you.

Comment (tbl3rd) [reply src/ptc/util/jms.clj:221]: I'll break this up if you'd rather but it's only 14 lines called from one place.

File: src/ptc/util/jms.clj (hunk 4 above)
[the remark this answers is not in the document]
Comment (tbl3rd) [reply src/ptc/util/jms.clj:162]: It's there to name the parameter for the docstring.
I'll add an `_` so the linter won't trip over it.

**Review by samanehsan — APPROVED** [review]
LGTM! Thanks for these improvements 🤩

**Review by ehigham — APPROVED** [review]

**Pull request merged**